### PR TITLE
Adds linting rules for ts* files

### PR DESCRIPTION
**Pull request by katerberg**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,6 @@
 language: "node_js"
 node_js:
-      - "node"
+  - "node"
+script:
+  - "npm run tslint"
```
Comment (katerberg) [inline .travis.yml]: we should have a `npm run prettier` in travis!

File: package.json [hunk 2]
```diff
@@ -6,7 +6,12 @@
   "scripts": {
     "prestart": "cp ./node_modules/react/umd/react.development.js dist/react.development.js && cp ./node_modules/react-dom/umd/react-dom.development.js dist/react-dom.development.js",
     "start": "webpack-dev-server --config ./webpack.config.js --mode development",
-    "test": "mocha -r ts-node/register src/**/*.test.ts"
+    "critter": "npm run tslint && npm test",
```
Comment (katerberg) [inline package.json]: Add prettier!

File: src/components/Hello.test.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
-import { Hello } from './Hello';
 import { expect } from 'chai';
 import 'mocha';
+import { Hello } from './Hello';
```
Comment (EmilyKraft) [inline src/components/Hello.test.ts]: why did this get moved? was this part of the linter?

File: package.json [hunk 4]
```diff
@@ -6,7 +6,14 @@
   "scripts": {
     "prestart": "cp ./node_modules/react/umd/react.development.js dist/react.development.js && cp ./node_modules/react-dom/umd/react-dom.development.js dist/react-dom.development.js",
     "start": "webpack-dev-server --config ./webpack.config.js --mode development",
-    "test": "mocha -r ts-node/register src/**/*.test.ts"
+    "critter": "npm run tslint && npm run prettier && npm test",
+    "test": "mocha -r ts-node/register src/**/*.test.ts",
```
Comment (tlj4k8) [inline package.json]: Is this needed? I believe we got rid of Mocha and are now using Jest?? @katerberg

File: src/components/Hello.test.ts (hunk 3 above)
Comment (tlj4k8) [reply src/components/Hello.test.ts]: I believe so... @katerberg Thoughts?

File: src/components/Hello.test.ts [hunk 5]
```diff
@@ -1,11 +1,11 @@
-import { Hello } from './Hello';
 import { expect } from 'chai';
```
Comment (tlj4k8) [inline src/components/Hello.test.ts]: Mocha is not USED!!!!!!!

File: src/components/Hello.test.ts [hunk 6]
```diff
@@ -1,11 +1,11 @@
...
 import 'mocha';
+import { Hello } from './Hello';
 
 describe('Hello', () => {
-    it('exists', () => {
-        const hello = new Hello({compiler: 'blah', framework: 'foo'});
+	it('exists', () => {
```
Comment (tlj4k8) [inline src/components/Hello.test.ts]: This whole test is incorrect. It is updated in master.

File: tslint.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{
+    "extends": ["tslint:latest", "tslint-react", "tslint-config-prettier"],
```
Comment (katerberg) [inline tslint.json]: this should use recommended

**Pull request merged**